.github/workflows/jsformat-lint.yml

```diff
@@ -0,0 +1,24 @@
+name: Format the JS code
+
+on: [pull_request]
+
+jobs:
+  format:
+    runs-on: ubuntu-latest
+    name: Format Files
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.head_ref }}
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "18.18.0"
+      - run: npm ci
+      - run: npm run prettier:format
+      - run: npm run lint:js
+      - name: Commit changes
+        uses: stefanzweifel/git-auto-commit-action@v4
+        if: ${{ github.event_name == 'push' || github.event_name == 'workflow_dispatch' }}
+        with:
+          commit_message: "github actions: style: format JS files"
+          branch: ${{ github.head_ref }}
```

Comment:
With lines 3 and 19-24, is this going to start an infinite loop where the PR gets updated so this code formatting runs and makes a commit, which is a PR update so the code formatter runs again? Or will it not commit the second time because nothing changed?

Do you need to be more specific about the PR actions that will trigger this workflow? https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request

Reply:
I checked this and it does not start infinite loop